### PR TITLE
remove unused code relate to old text-input component

### PR DESCRIPTION
**Pull request by qfrank**
when i was trying to move mention implementation to `status-go` , i found we have 2 components for input:
```
src/status_im/ui/screens/chat/components/input.cljs
src/status_im/ui2/screens/chat/composer/input.cljs
```
the latter one is a new implementation, `src/status_im/ui/screens/chat/components/input.cljs` can be safely removed now. 

### Testing notes
small refactor relate to mention has been included in this PR. edit message should works as before.

#### Platforms

- Android
- iOS

status: ready

### CODE REVIEW

File: src/status_im/chat/models/mentions.cljs [hunk 1]
```diff
@@ -223,3 +236,5 @@
      (native-module/reset-keyboard-input
       (rn/find-node-handle (react/current-ref ref))
       cursor))))
+
+
```
Comment (rasom) [inline src/status_im/chat/models/mentions.cljs]: Do we really need those tow lines?

**Review by rasom — APPROVED** [review]

File: src/status_im/chat/models/mentions.cljs (hunk 1 above)
Comment (qfrank) [reply src/status_im/chat/models/mentions.cljs]: thank you, fixed 🙂

**Review by flexsurfer — APPROVED** [review]

**Review by J-Son89 — APPROVED** [review]
Nice!

**Comment by pavloburykh** [comment]
@qfrank thanx for the PR. Please take a look at the following issue

### ISSUE 1 Mention text overlays message text when mention is added during message editing (Android)

Reproducible only for Android

**Steps:** 

1. Open any chat
2. Send a message
3. Edit message: add mention during editing
4. See how message looks after editing 

**Actual result:** mention overlays message text. It is fixed after chat reopening. 

photo_2023-04-27 15 51 06

https://user-images.githubusercontent.com/97245802/234867338-932c3af0-9b00-4988-9bb6-f85cad64e7ca.mp4

**Comment by qfrank** [comment]
> @qfrank thanx for the PR. Please take a look at the following issue
> 
> ### ISSUE 1 Mention text overlays message text when mention is added during message editing (Android)
> Reproducible only for Android
> 
> **Steps:**
> 
> 1. Open any chat
> 2. Send a message
> 3. Edit message: add mention during editing
> 4. See how message looks after editing
> 
> **Actual result:** mention overlays message text. It is fixed after chat reopening.
> 
> photo_2023-04-27 15 51 06
> 
>  telegram-cloud-document-2-5357046119127853093.mp4

Hi @pavloburykh , can you check if this issue also exist on latest `develop` branch?  if it does, it won't be fixed in this PR

**Comment by pavloburykh** [comment]
> Hi @pavloburykh , can you check if this issue also exist on latest `develop` branch? if it does, it won't be fixed in this PR

My bad @qfrank. It exists in develop. Very weird, as it has been recently fixed. Will log this issue separately then.

**Comment by pavloburykh** [comment]
@qfrank PR is ready to be merged. Thank you!

**Pull request merged**